packages/functional-tests/tests/react-conversion/legal.spec.ts

```diff
@@ -6,53 +6,102 @@ import { test, expect } from '../../lib/fixtures/standard';
+      await expect(privacy.backButton).toBeVisible();
+
+      await privacy.clickBackButton();
+
+      await expect(legal.pageHeader).toBeVisible();
+    });
+
+    test('users can navigate from privacy notice to legal from third-party', async ({
+      page,
+      pages: { legal, privacy },
+    }) => {
+      test.fixme(true, 'Fix required as of 2024/03/13 (see FXA-9280).');
+
+      await page.goto(`https://www.mozilla.org/en-US/`);
+      await privacy.goto();
+
+      await expect(privacy.pageHeader).toBeVisible();
+      await expect(privacy.backButton).toBeVisible();
+
+      await privacy.clickBackButton();
+
+      await expect(legal.pageHeader).toBeVisible();
     });
 
-    test('start at terms page', async ({ page, target }) => {
-      await page.goto(`${target.contentServerUrl}/legal/terms`);
+    test('users can navigate between legal and terms of service', async ({
+      pages: { legal, termsOfService },
+    }) => {
+      await legal.goto();
+
+      await expect(legal.pageHeader).toBeVisible();
+      await expect(legal.termsOfServiceLink).toBeVisible();
+
+      await legal.clickTermsOfServiceLink();
 
-      // Verify legal page is visible
-      // this text is not in our codebase, it's pulled from the `legal-docs` repo
-      await page
-        .locator('text="Firefox Cloud Services: Terms of Service"')
-        .isVisible();
+      await expect(termsOfService.pageHeader).toBeVisible();
+
+      await termsOfService.clickBackButton();
+
+      await expect(legal.pageHeader).toBeVisible();
     });
 
-    test('start at privacy page', async ({ page, target }) => {
-      await page.goto(`${target.contentServerUrl}/legal/privacy`);
+    test('users can navigate from terms of service to legal', async ({
+      pages: { legal, termsOfService },
+    }) => {
+      test.fixme(true, 'Fix required as of 2024/03/13 (see FXA-9280).');
+
+      await termsOfService.goto();
+
+      await expect(termsOfService.pageHeader).toBeVisible();
+      await expect(termsOfService.backButton).toBeVisible();
+
+      await termsOfService.clickBackButton();
+
+      await expect(legal.pageHeader).toBeVisible();
+    });
+
+    test('users can navigate from terms of service to legal from third-party', async ({
+      page,
+      pages: { legal, termsOfService },
+    }) => {
+      test.fixme(true, 'Fix required as of 2024/03/13 (see FXA-9280).');
+
+      await page.goto(`https://www.mozilla.org/en-US/`);
+      await termsOfService.goto();
+
+      await expect(termsOfService.pageHeader).toBeVisible();
+      await expect(termsOfService.backButton).toBeVisible();
+
+      await termsOfService.clickBackButton();
 
-      // Verify privacy page is visible
-      await page.waitForTimeout(1000);
-      // this text is not in our codebase, it's pulled from the `legal-docs` repo
-      await page.locator('text="Firefox Privacy Notice"').isVisible();
+      await expect(legal.pageHeader).toBeVisible();
```

Comment:
Thanks for changing to expect().tobeVisible()  :)

Reply:
I also remember this page takes a while to be displayed hence a waitForTimeout() was introduced but yes that's a problem when we fix the back button and start running these tests again.

Reply:
I am hoping that by replacing `await page.waitForTimeout(1000);` with a web first assertion, that we will have better results.